### PR TITLE
Retriever update in repository fails

### LINKED ISSUE #322: Running `retriever update` from inside the repo fails
We wouldn't normally do this since the scripts are all in the repo, but this can cause confusion (e.g., https://github.com/weecology/retriever/pull/320#issuecomment-101162033). This should either work or fail explicitly with instructions to not use `update` from inside the repo.

```
ethan@oryx:~/RetrieverCode/src (master)$ retriever update
0.0-bbs.py 85.1-bbs.py 100.0-bbs.py 0.0-EA_del_moral_2010.script 100.0-EA_del_moral_2010.script 0.0-npn.py 100.0-npn.py 0.0-bbs50stop.py 72.7-bbs50stop.py 100.0-bbs50stop.py 0.0-EA_clark2006.py 100.0-EA_clark2006.py 0.0-gentry.py 77.9-gentry.py 100.0-gentry.py 0.0-EA_portal_mammals.script 100.0-EA_portal_mammals.script 0.0-EA_petraitis2008.py 100.0-EA_petraitis2008.py [Errno 2] No such file or directory: 'scripts/bbs.py'
[Errno 2] No such file or directory: 'scripts/EA_del_moral_2010.script'
[Errno 2] No such file or directory: 'scripts/npn.py'
[Errno 2] No such file or directory: 'scripts/bbs50stop.py'
[Errno 2] No such file or directory: 'scripts/EA_clark2006.py'
[Errno 2] No such file or directory: 'scripts/gentry.py'
[Errno 2] No such file or directory: 'scripts/EA_portal_mammals.script'
[Errno 2] No such file or directory: 'scripts/EA_petraitis2008.py'
```

### PR DESCRIPTION
**Pull request by henrykironde**
fixes #322

### CODE REVIEW

File: lib/repository.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ def download_from_repository(filepath, newpath, repo=REPOSITORY):
         def reporthook(a,b,c):
             print "%3.1f-%s" % (min(100, float(a * b) / c * 100), filename),
             sys.stdout.flush()
+            print
```
Comment (ethanwhite) [inline lib/repository.py]: Is this `print` accomplishing something?

File: lib/repository.py [hunk 2]
```diff
@@ -76,6 +78,7 @@ def flush(self):
     if graphical:
         splash.Hide()
         sys.stdout = sys.__stdout__
+    print "retriever up to date"
```
Comment (ethanwhite) [inline lib/repository.py]: "retriever up to date" -> "The retriever is up-to-date"

File: lib/repository.py [hunk 3]
```diff
@@ -156,33 +162,35 @@ def run(self):
...
                     print "Downloading script: " + script_name
                     download_from_repository("scripts/" + script_name,
-                                             os.path.join(SCRIPT_WRITE_PATH, script_name))
-                elif script_version:
-                    # File exists: import and check MD5 sum
-                    need_to_download = False
+                                             os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)))
+
+                # check MD5sum based on the script version to download the right scripts
+                # if MD5sum dont match; need_to_download is True
```
Comment (ethanwhite) [inline lib/repository.py]: "if MD5sum dont match; need_to_download is True" -> "if the MD5sum doesn't match need_to_download is set to True"

**Comment by ethanwhite** [comment]
Works great! Once the minor documentation changes are made this is ready to go.

File: lib/repository.py (hunk 1 above)
Comment (henrykironde) [inline lib/repository.py]: retriever update results
0.0-bbs.py 85.1-bbs.py 100.0-bbs.py 0.0-EA_del_moral_2010.script 100.0-EA_del_moral_2010.script 0.0-npn.py 100.0-npn.py 0.0-bbs50stop.py 72.7-bbs50stop.py 100.0-bbs50stop.py 0.0-EA_clark2006.py 1
added new live to print each script on a new line (removed the print and include newline in  'print "%3.1f-%s\n" %'

File: lib/repository.py (hunk 2 above)
Comment (henrykironde) [inline lib/repository.py]: done

File: lib/repository.py (hunk 3 above)
Comment (henrykironde) [inline lib/repository.py]: covered

**Comment by ethanwhite** [comment]
Looks great!

**Comment by ethanwhite** [comment]
Merged in 4b24063278b54bd46fe70851802d4f528361508f.

**Pull request closed without merging**